src/core/types/addonTypes.js

```diff
@@ -0,0 +1,82 @@
+/* @flow */
+import {
+  ADDON_TYPE_DICT,
+  ADDON_TYPE_EXTENSION,
+  ADDON_TYPE_LANG,
+  ADDON_TYPE_SEARCH,
+  ADDON_TYPE_THEME,
+} from 'core/constants';
+
+export type AddonVersionType = {|
+  id: number,
+  channel: string,
+  edit_url: string,
+  files: Array<Object>,
+  // The `text` property is omitted from addon.current_version.license.
+  license: { name: string, url: string },
+  reviewed: Date,
+  version: string,
+|};
+
+export type AddonAuthorType = {|
+  name: string,
+  url: string,
+|};
+
+export type AddonTypeProp =
+  typeof ADDON_TYPE_DICT |
```

Comment:
These look like they should be `||` `OR`s not bitwise `OR`s, as I assume we want `string` here not `0`. Odd that the type being `0` didn't mess stuff up somewhere...